packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -524,6 +536,9 @@ typedef void _DeviceEventHandler(Device device);
 /// `device.removed` events.
 class DeviceDomain extends Domain {
   DeviceDomain(Daemon daemon) : super(daemon, 'device') {
+    if (!doctor.canListAnything) {
+      daemon.shutdown(error: new ToolExit('Unable to locate a development device', exitCode: 1));
```

Comment:
Can we say anything more specific here? For example, this would be a great time to show ANDROID_HOME.